src/api/app/models/workflow/step/branch_package_step.rb

```diff
@@ -16,24 +16,27 @@ def target_project_base_name
   end
 
   def branch_package(workflow_filters = {})
-    source_project = Project.get_by_name('source_project_name')
-    source_package = Package.get_by_project_and_name(source_project_name, source_package_name)
+    scmsync = false
     
-    if source_project.scmsync.present? || source_package.scmsync.present?
-      add_scmsync_to_meta(package: target_package, scmsync_url: source_package.scmsync) if source_package.scmsync.present?
-      add_scmsync_to_meta(package: target_package, scmsync_url: source_project.scmsync, scmsynced_project: true) if source_project.scmsync.present?
-      # TODO: set repositories on target_package
-    else
-      create_branched_package if scm_webhook.new_pull_request? || (scm_webhook.updated_pull_request? && target_package.blank?) || scm_webhook.push_event? || scm_webhook.tag_push_event?
-      add_branch_request_file(package: target_package)
-    end
+    # if the whole project is scm synced we find the `Project` object but no `Package` objects (they only exist in the backend)
+    # if the scmsync is setup on the package level, the `Package` object exists
+    source_project = Project.get_by_name(source_project_name) 
+    source_package = Package.get_by_project_and_name(source_project_name, source_package_name)
```

Comment:
This will be [nil](https://github.com/krauselukas/open-build-service/blob/eb3d0f40ad6110d6b76c4d139649220788e30eb7/src/api/app/models/package.rb#L167)